src/server/scripts/Shadowlands/Sanctum of Domination/boss_sylvanas_windrunner.cpp

```diff
@@ -0,0 +1,3694 @@
+        Position arrowInnerRight(arrowCenter.GetPositionX() + (std::cos(orientation + -135.0f * M_PI / 180) * 2.8284f), arrowCenter.GetPositionY() + (std::sin(orientation + -135.0f * M_PI / 180) * 2.8284f), arrowCenter.GetPositionZ());
+
+        scheduler.Schedule(Milliseconds(step * 10 + 50), [this, arrowInnerLeft, arrowInnerRight](TaskContext context)
+        {
+            me->CastSpell(arrowInnerLeft, SPELL_DESECRATING_SHOT_AREATRIGGER, true);
+            me->CastSpell(arrowInnerRight, SPELL_DESECRATING_SHOT_AREATRIGGER, true);
+        });
+
+        Position arrowOuterLeft(arrowCenter.GetPositionX() + (std::cos(orientation + 135.0f * M_PI / 180) * 5.6568f), arrowCenter.GetPositionY() + (std::sin(orientation + 135.0f * M_PI / 180) * 5.6568f), arrowCenter.GetPositionZ());
+        Position arrowOuterRight(arrowCenter.GetPositionX() + (std::cos(orientation + -135.0f * M_PI / 180) * 5.6568f), arrowCenter.GetPositionY() + (std::sin(orientation + -135.0f * M_PI / 180) * 5.6568f), arrowCenter.GetPositionZ());
+
+        scheduler.Schedule(Milliseconds(step * 10 + 75), [this, arrowOuterLeft, arrowOuterRight](TaskContext context)
+        {
+            me->CastSpell(arrowOuterLeft, SPELL_DESECRATING_SHOT_AREATRIGGER, true);
+            me->CastSpell(arrowOuterRight, SPELL_DESECRATING_SHOT_AREATRIGGER, true);
+        });
+
+        return true;
+    }
+
+    void DrawDesecratingShotScattered()
+    {
+        for (uint8 i = 0; i < 8; i++)
+        {
+            for (uint8 i = 0; i < 2; i++)
+            {
+                scheduler.Schedule(250ms, [this](TaskContext context)
+                {
+                    me->CastSpell(me->GetRandomPoint(me->GetPosition(), frand(0.0f, 40.0f)), SPELL_DESECRATING_SHOT_AREATRIGGER, CastSpellExtraArgs(TRIGGERED_FULL_MASK).AddSpellMod(SPELLVALUE_DURATION, 3000));
+                });
+            }
+        }
+
+        for (uint8 desecratingShotOnPlayer = std::max<uint8>(4, std::ceil(float(me->GetMap()->GetPlayersCountExceptGMs()) / 3)) / 3.0f; desecratingShotOnPlayer > 0; desecratingShotOnPlayer--)
+        {
+            if (Unit* target = SelectTarget(SELECT_TARGET_RANDOM, 0, 150.0f, true, true))
+            {
+                for (uint8 i = 0; i < 2; i++)
+                {
+                    scheduler.Schedule(250ms, [this, target](TaskContext context)
+                    {
+                        me->CastSpell(target->GetRandomPoint(target->GetPosition(), frand(1.0f, 2.5f)), SPELL_DESECRATING_SHOT_AREATRIGGER, CastSpellExtraArgs(TRIGGERED_FULL_MASK).AddSpellMod(SPELLVALUE_DURATION, 3000));
+                    });
+                }
+            }
+        }
+    }
+
+    void DrawDesecratingShotSpiral()
+    {
+        // Three casts
+    }
+
+    void DrawDesecratingShotBentArrow()
+    {
+
+    }
+
+    void DrawDesecratingShotJar()
+    {
+
+    }
+
+    ObjectGuid GetShadowCopyJumperGuid()
```

Comment:
you may want to just add a parameter here: `int32 index` and then `return _shadowCopyGUID[index]`, that way you do not require 2 separate functions and can use the other guids also very quickly without writing a new method for each

Reply:
added on next commit.